feat: 프로젝트 설정

scarf005 opened the pull request:
## 개요

- fixes #1 

### Swagger UI

https://github.com/7rd-wanted-pre-onboarding-team-H/w1-social-feed/assets/54838975/ada43f88-38d1-4da2-a9ce-b98eec8eaca3

### 앱 실행 및 테스트

https://github.com/7rd-wanted-pre-onboarding-team-H/w1-social-feed/assets/54838975/0d179706-62a9-40a9-aadc-f5e1a781c608

- sqlite3 설정
- DB migrate(up, down) 구현
- openapi 및 swagger 추가
- in-memory DB 사용 테스트 추가

## TODO

- [x] 사용법 및 추가 설명 필요
- [x] 커밋 정리 필요
- [x] 라우터별로 정리

u00938 commented on main.ts:
```diff
@@ -0,0 +1,46 @@
+			return c.jsonT({ message: `hello ${name}` })
+		})
+		.openapi(getPosting, async (c) => {
+			const { id } = c.req.valid("param")
```
이 라우트들도 따로 뺄 수 있으면 더 좋겠지만 현재 구조에서는 일단 이 비즈니스 로직 부분도 따로 서비스 폴더에서 진행하는 게  좋을 것 같습니다
(예시: service/user, posting)

scarf005 replied:
넵, 별도 폴더로 분리하도록 하겠습니다.

scarf005 committed:
feat: deno 설정

포매터, 테스트, 사용  패키지 목록 설정을 추가했습니다.

scarf005 committed:
chore: vscode 설정 추가

개발에 참고할 수 있는 settings.example.json을 추가했습니다.

scarf005 committed:
docs: 실행 및 개발 가이드 추가

Commit:
build: kysely 타입 생성 패키지 추가

- `deno task db:gen-types` 명령어로 타입을 생성합니다.
- `deno task db:init`으로 `test.db`를 초기화합니다.
- `test.db`에서 `types.ts`와 `schemas.ts`를 추가합니다.

scarf005 committed:
feat: DB 마이그레이션 함수

scarf005 committed:
feat: DB 더미 값 생성

`deno task db:seed`를 통해 DB를 더미 값으로 채울 수 있습니다.

scarf005 committed:
feat: kyselyFrom

주어진 DB 경로에서 kysely 인스턴스를 생성하는 유틸리티 함수입니다

scarf005 committed:
feat: Swagger UI 렌더링 함수

scarf005 committed:
feat: util 함수

테스트와 라우트 생성 헬퍼 함수들입니다.

scarf005 committed:
feat: `/hello` 엔드포인트

예시 엔드포인트입니다.

scarf005 committed:
feat: `/posting/{id}`

게시물 하나를 DB에서 쿼리하는 예시 엔드포인트입니다.

scarf005 committed:
feat: main

메인 서버입니다. `deno task dev`로 실행 가능합니다.

scarf005 committed:
ci: 자동 테스트

scarf005 committed:
style: 포매팅

scarf005 committed:
docs: 오타 수정

jimin52 reviewed (approved):
고생하셨습니다. 주석과 README 가 상세해서 좋습니다. 머지 후 로컬에서 돌려볼게요
